pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -575,16 +574,26 @@ func publishMetrics(ctx *zedagentContext, iteration int) {
 	log.Debugf("publishMetrics: after send, total elapse sec %v", time.Since(startPubTime).Seconds())
 }
 
-func getDiskInfo(diskfile string, appDiskDetails *metrics.AppDiskMetric) error {
-	imgInfo, err := diskmetrics.GetImgInfo(diskfile)
-	if err != nil {
-		return err
+func getDiskInfo(ss types.StorageStatus, appDiskDetails *metrics.AppDiskMetric) error {
+	if ss.IsContainer {
+		appDiskDetails.Disk = ss.ActiveFileLocation
+		// XXX For container images, max size is coming zero
+		// from the controller. So for now, we are setting up
+		// total size equal to the used size.
+		appDiskDetails.Provisioned = RoundToMbytes(ss.Size)
```

Comment:
If we had a non-zero MaxSize aka Size in the config, maybe we should report that here as the Provisioned size?
(Separately we'd need to figure out a way to enforce such a max size on a container.)

Reply:
Makes sense. If MaxSize is non-zero in the status, we should report that to the controller. I will update the PR.

Reply:
>(Separately we'd need to figure out a way to enforce such a max size on a container.)

I think we don't need to do any work on this part because container bootable disk will always have volumes attached and those volumes will have proper MaxSize. For container bootable disk, both Provisioned and Used size will be the same.